### PR TITLE
[CAPT-604] Update azurerm terraform provider

### PR DESCRIPTION
**Pull request by saliceti**
## What
Update azurerm provider to clear some tech debt and hope it will solve some small ramdom errors we've experienced

## How to review
Successful run: https://dev.azure.com/dfe-ssp/S118-Teacher-Payments-Service/_build/results?buildId=286267&view=results
Review app: https://s118d02-app-pr-2095-as.azurewebsites.net/additional-payments/landing-page

## ⚠️ Important
As the resource is replaced, it will create a small downtime the first time it is applied to an environment

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -34,7 +34,7 @@ set-azure-account:
 
 terraform-init: set-azure-account
 	$(if $(IMAGE_TAG), , $(error Missing environment variable "IMAGE_TAG"))
-	terraform -chdir=azure/terraform init -reconfigure \
+	terraform -chdir=azure/terraform init -reconfigure -upgrade \
```
Comment (welshman1970) [inline Makefile:37]: where does it get the upgrade file?

File: azure/terraform/modules/app_service/app_service.tf [hunk 2]
```diff
@@ -1,78 +1,15 @@
-resource "azurerm_app_service" "app_as" {
+resource "azurerm_linux_web_app" "app_as" {
```
Comment (welshman1970) [inline azure/terraform/modules/app_service/app_service.tf:1]: Why has it changed from azurerm_app_service to azurerm_linux_web_app

File: azure/terraform/modules/app_service/app_service.tf [hunk 3]
```diff
@@ -1,78 +1,15 @@
...
   name                = local.app_service_name
   resource_group_name = var.app_rg_name
   location            = var.rg_location
-  app_service_plan_id = data.azurerm_app_service_plan.app.id
+  service_plan_id     = data.azurerm_service_plan.app.id
```
Comment (welshman1970) [inline azure/terraform/modules/app_service/app_service.tf:5]: why has this changed?

**Review by welshman1970 — COMMENTED** [review]
just a few comments regarding some changes to some of the resources

File: azure/terraform/modules/app_service/app_service.tf (hunk 2 above)
Comment (saliceti) [reply azure/terraform/modules/app_service/app_service.tf:1]: azurerm_app_service has been deeprecated in the new provider version

File: azure/terraform/modules/app_service/app_service.tf (hunk 3 above)
Comment (saliceti) [reply azure/terraform/modules/app_service/app_service.tf:5]: service_plan_id has been renamed in the new provider version

**Review by slorek — APPROVED** [review]

**Pull request merged**